qemu/tests/qemu_guest_agent.py

```diff
@@ -1723,17 +1729,33 @@ def _change_bl(self, session):
         for windows guest.
         """
         if self.params.get("os_type") == "linux":
-            cmd_black_list = self.params["black_list"]
-            cmd_blacklist_backup = self.params["black_list_backup"]
+            cmd_blacklist_backup = self.params["black_file_backup"]
             session.cmd(cmd_blacklist_backup)
-            for black_cmd in cmd_black_list.split():
-                bl_check_cmd = self.params["black_list_check_cmd"] % black_cmd
-                bl_change_cmd = self.params["black_list_change_cmd"] % black_cmd
+            full_qga_ver = self._get_main_qga_version(session, self.vm, main_ver=False)
+            full_qga_ver = version.parse(full_qga_ver)
+            if full_qga_ver >= version.parse("8.1.0-5"):
+                cmd_black_list = self.params["black_list_new"]
+                bl_check_cmd = self.params["black_list_check_cmd"] % cmd_black_list
+                bl_change_cmd = self.params["black_list_change_cmd"] % cmd_black_list
                 session.cmd(bl_change_cmd)
                 output = session.cmd_output(bl_check_cmd)
-                if not output == "":
+                if output == "":
                     self.test.fail("Failed to change the cmd to "
                                    "white list, the output is %s" % output)
+            else:
+                cmd_black_list = self.params["black_list"]
+                for black_cmd in cmd_black_list.split():
+                    bl_check_cmd = self.params["black_list_check_cmd"] % black_cmd
+                    if (self.params["gagent_check_type"] == "blacklist" and
+                            full_qga_ver >= version.parse("7.2.0-4")):
+                        bl_change_cmd = self.params["black_list_change_cmd"]
+                    else:
+                        bl_change_cmd = self.params["black_list_change_cmd"] % black_cmd
+                    session.cmd(bl_change_cmd)
+                    output = session.cmd_output(bl_check_cmd)
+                    if not output == "":
+                        self.test.fail("Failed to change the cmd to "
+                                       "white list, the output is %s" % output)
```

Comment:
I believe this `if...else` block could be optimized, I will try to twist it and update here later.